training_record/services.py

```diff
@@ -134,6 +141,9 @@ def update_off_campus_record_from_raw_data(
                 setattr(off_campus_event_instance, key, val)
             off_campus_event_instance.save()
 
+            record.event_coefficient.role = role
```

Comment:
According to the official doc:

If you’re just updating a record and don’t need to do anything with the model object, the most efficient approach is to call update(), rather than loading the model object into memory. For example, instead of doing this:
```
e = Entry.objects.get(id=10)
e.comments_on = False
e.save()
```
do this:
```
Entry.objects.filter(id=10).update(comments_on=False)
```
So in your case, I think you can use:
```
EventCoefficient.objects.filter(id=record.event_coefficient_id).update(role=role)
```